### PR TITLE
Expends contribution info table

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/6942

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/brave_rewards/browser/database/database_contribution_info.cc [hunk 1]
```diff
@@ -13,14 +13,28 @@
 #include "sql/statement.h"
 #include "sql/transaction.h"
 
+namespace {
+  double ProbiToDouble(const std::string& probi) {
+    const size_t size = probi.size();
+    std::string amount = "0";
+    if (size > 18) {
```
Comment (masparrow) [inline components/brave_rewards/browser/database/database_contribution_info.cc]: const for `18` as used twice

File: components/brave_rewards/browser/database/database_util.cc [hunk 2]
```diff
@@ -9,6 +9,18 @@
 
 namespace brave_rewards {
 
+bool DropTable(
+    sql::Database* db,
+    const std::string& table_name) {
+  DCHECK(!table_name.empty());
```
Comment (masparrow) [inline components/brave_rewards/browser/database/database_util.cc]: We should check `db` as well - I'm guessing this pattern (of not checking) is well established - but I'd recommend adopting it going forward.

File: components/brave_rewards/browser/database/publisher_info_database_unittest.cc [hunk 3]
```diff
@@ -1106,27 +1108,23 @@ TEST_F(PublisherInfoDatabaseTest, Migrationv7tov8_ContributionInfo) {
...
   EXPECT_TRUE(publisher_info_database_->Init());
 
-  ContributionInfo contribution;
-  contribution.probi = "1000000000000000000";
-  contribution.month = static_cast<int>(ledger::ActivityMonth::OCTOBER);
-  contribution.year = 2019;
-  contribution.type = static_cast<int>(ledger::RewardsType::ONE_TIME_TIP);
-  contribution.date = 1570614352;
-  contribution.publisher_key = "3zsistemi.si";
+  const std::string publisher_key = "3zsistemi.si";
```
Comment (masparrow) [inline components/brave_rewards/browser/database/publisher_info_database_unittest.cc]: Question here - I recognise this string but is it 'owned' by Brave?

Comment (NejcZdovc) [reply components/brave_rewards/browser/database/publisher_info_database_unittest.cc]: that's my domain 😃

Comment (masparrow) [reply components/brave_rewards/browser/database/publisher_info_database_unittest.cc]: Whups - was thinking 'integration' test - not unit! Discarding this comment.

File: components/brave_rewards/browser/database/database_util.cc (hunk 2 above)
Comment (NejcZdovc) [reply components/brave_rewards/browser/database/database_util.cc]: done

File: components/brave_rewards/browser/database/database_contribution_info.cc (hunk 1 above)
Comment (NejcZdovc) [reply components/brave_rewards/browser/database/database_contribution_info.cc]: done

File: vendor/bat-native-ledger/src/bat/ledger/internal/bat_util.cc [hunk 4]
```diff
@@ -29,4 +29,15 @@ std::string ConvertToProbi(const std::string& amount) {
   return before_dot + after_dot + rest_probi;
 }
 
+double ProbiToDouble(const std::string& probi) {
+  const size_t size = probi.size();
+  std::string amount = "0";
+  if (size > 18) {
+    amount = probi;
```
Comment (masparrow) [inline vendor/bat-native-ledger/src/bat/ledger/internal/bat_util.cc]: const int32_t probi_size = 18;

(as per database_contribution.cc) - perhaps this should be a more accessible const somewhere?

Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/bat_util.cc]: it's temporary thing while we are migrating to double. Probi will go away as soon as possible

**Pull request merged**